Analysis/src/SignalObjectSelector.cpp

```
@@ -108,8 +109,13 @@ bool SignalObjectSelector::PassLeptonVetoSelection(const ntuple::Event& event) c
           if(eleId_iso.Passed(DiscriminatorWP::Medium)) return false;
         }
         if(static_cast<LegType>(event.other_lepton_type.at(n)) == LegType::mu){
-          analysis::DiscriminatorIdResults muonId(event.other_lepton_muonId.at(n));
-          if(muonId.Passed(DiscriminatorWP::Medium)) return false;
+            const Channel channel = static_cast<Channel>(event.channelId);
+            if(channel == Channel::MuTau){ //this is a temporary fix because of an error in the production code for vetoMuons
+                if(static_cast<LegType>(event.lep_type.at(0)) != LegType::mu) continue;
```

throw an exception instead of continue